userdocs/diagnostics/string-interpolation-conformance.md

```diff
@@ -0,0 +1,10 @@
+# Conforming to `StringInterpolationProtocol`
+Types can implement custom interpolation behavior in string literals by conforming to `StringInterpolationProtocol`. In addition to its formal requirements, `init(literalCapacity:interpolationCount:)` and `appendLiteral(_:)`, `StringInterpolationProtocol` has an additional, informal requirement, `appendInterpolation`. String interpolations using `\()` syntax are translated into calls to matching `appendInterpolation` methods.
+
+`StringInterpolationProtocol` conformers must provide at least one valid `appendInterpolation` method, but it can have any number of overloads. A valid method must:
+
+- Be marked `mutating`
+- Not be marked `static`
+- Either return `Void` or be marked `@discardableResult`
+
+To learn more about customizing string interpolation behavior, see the standard library documentation of the [`ExpressibleByStringInterpolation`](https://developer.apple.com/documentation/swift/expressiblebystringinterpolation) and [`StringInterpolationProtocol`](https://developer.apple.com/documentation/swift/stringinterpolationprotocol) protocols.
```

Comment:
I'm not sure these links are stable, but this is the only hosted documentation for these protocols I know of

Reply:
Perhaps some of the SourceKit folks will have some ideas about how to handle URLs like this. Paging @nathawes.

Reply:
Not sure myself, sorry. 
@amigi do you know if these are the best links to use / who to ask?

Reply:
I will check and follow up. Thanks for asking!